scripts/searchindex.ts

```diff
@@ -0,0 +1,211 @@
+const dotenv = require('dotenv');
+
+dotenv.config({ path: '.env.elastic' }); // ignore errors
+
+const mappings = {
+  dynamic: 'false',
+  properties: {
+    body_content: {
+      type: 'text',
+      term_vector: 'with_positions_offsets',
+      fields: {
+        raw: {
+          type: 'keyword',
+        },
+      },
+    },
+    id: {
+      type: 'keyword',
+    },
+    date: {
+      type: 'date',
+    },
+    tags: {
+      type: 'text',
+      fields: {
+        raw: {
+          type: 'keyword',
+        },
+      },
+    },
+    title: {
+      type: 'text',
+      fields: {
+        raw: {
+          type: 'keyword',
+        },
+      },
+    },
+    summary: {
+      type: 'text',
+      fields: {
+        raw: {
+          type: 'keyword',
+        },
+      },
+    },
+    url: {
+      type: 'text',
+      fields: {
+        raw: {
+          type: 'keyword',
+        },
+      },
+    },
+    url_host: {
+      type: 'text',
+      fields: {
+        raw: {
+          type: 'keyword',
+        },
+      },
+    },
+    url_path: {
+      type: 'text',
+      fields: {
+        raw: {
+          type: 'keyword',
+        },
+      },
+    },
+    url_path_dir1: {
+      type: 'text',
+      fields: {
+        raw: {
+          type: 'keyword',
+        },
+      },
+    },
+    url_path_dir2: {
+      type: 'text',
+      fields: {
+        raw: {
+          type: 'keyword',
+        },
+      },
+    },
+    url_path_dir3: {
+      type: 'text',
+      fields: {
+        raw: {
+          type: 'keyword',
+        },
+      },
+    },
+    url_port: {
+      type: 'text',
+      fields: {
+        raw: {
+          type: 'keyword',
+        },
+      },
+    },
+  },
+};
+
+const client = new Client({
+  node: {
+    url: new URL(process.env.ELASTIC_URL),
+  },
+  auth: {
+    username: process.env.ELASTIC_USERNAME,
+    password: process.env.ELASTIC_PASSWORD,
+  },
+  maxRetries: 5,
+  requestTimeout: 60000,
+  sniffOnStart: true,
+});
+
+function getCurrentDate(): string {
+  const now = new Date();
+  const year = now.getFullYear().toString().substr(-2);
+  const month = (now.getMonth() + 1).toString().padStart(2, '0');
+  const day = now.getDate().toString().padStart(2, '0');
+  const hours = now.getHours().toString().padStart(2, '0');
+  const minutes = now.getMinutes().toString().padStart(2, '0');
+  return `${month}${year}${day}${hours}${minutes}`;
+}
+
+(async () => {
+  const { allBlogs } = await import('../.contentlayer/generated/index.mjs');
+  const { execa } = await import('execa');
+
+  const index = `site-search-dev-blog-${getCurrentDate()}`;
+
+  try {
+    await client.indices.create({
+      index,
+    });
+  } catch (e) {
+    if (!e.message.includes('resource_already_exists_exception')) {
+      throw e;
+    }
+  }
+
+  await client.indices.putMapping({
+    index,
+    body: mappings,
+  });
+
+  allBlogs.forEach(async (blog) => {
+    const body = blog.body.raw.replace(/<\/?[^>]+(>|$)/g, '');
```

Comment:
Just a note:

There might be components we want to strip from the indexed data, like [NewsletterRepublishing](https://github.com/tailscale-dev/tailscale-dev/blob/main/components/newsletter-republishing.tsx)

Reply:
This strips all components

Reply:
I was referring more to the content of a particular component. Even creating a "<noindex />" component, so the content it wraps is not indexed.